### PR TITLE
fix check for tenantId

### PR DESCRIPTION
**Pull request by jennyf19**
#1795

### CODE REVIEW

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @jennyf19

File: test/Microsoft.IdentityModel.Validators.Tests/MicrosoftIdentityIssuerValidatorTest.cs [hunk 1]
```diff
@@ -305,6 +306,31 @@ public void Validate_IssuerNotInTokenValidationParameters_ReturnsIssuer(string t
...
+            var context = new CompareContext();
+            var validator = new AadIssuerValidator(_httpClient, issuer);
+            var tidClaim = new Claim(tidClaimType, ValidatorConstants.TenantIdAsGuid);
+
+            var issClaim = new Claim(ValidatorConstants.ClaimNameIss, issuer);
+            List<Claim> claims = new List<Claim>();
+            claims.Add(tidClaim);
+            claims.Add(issClaim);
+
+            var jsonWebToken = new JsonWebToken(Default.Jwt(Default.SecurityTokenDescriptor(Default.SymmetricSigningCredentials, claims)));
```
Comment (GeoK) [inline test/Microsoft.IdentityModel.Validators.Tests/MicrosoftIdentityIssuerValidatorTest.cs:325]: Can we use one of the JsonWebTokenHandler.CreateToken (perhaps the one that takes SecurityTokenDescriptor) methods instead and avoid adding SecurityTokenDescriptor(credentials, claims)?

File: src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs [hunk 2]
```diff
@@ -264,7 +264,7 @@ private static string GetTenantIdFromToken(SecurityToken securityToken)
                 if (jsonWebToken.TryGetPayloadValue(AadIssuerValidatorConstants.Tid, out string tid))
                     return tid;
 
-                if (jsonWebToken.TryGetPayloadValue(AadIssuerValidatorConstants.Tid, out string tenantId))
+                if (jsonWebToken.TryGetPayloadValue(AadIssuerValidatorConstants.TenantId, out string tenantId))
```
Comment (dannybtsai) [inline src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs:267]: >TenantId [](http://example.com/codeflow?start=80&length=8)

Was the old code a mistake? It was getting Tid twice, as tid and tenantId.

File: test/Microsoft.IdentityModel.Validators.Tests/MicrosoftIdentityIssuerValidatorTest.cs (hunk 1 above)
Comment (jennyf19) [reply test/Microsoft.IdentityModel.Validators.Tests/MicrosoftIdentityIssuerValidatorTest.cs:325]: I tried to do that, but I don't want an encrypted token and also needed some custom values: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1801/files#diff-20c5742d81614f58ad5fd9e5ccc34931d030cbaf5b182bfbc8607707aa3a275bR811 and the claims were not getting added. thoughts?

File: src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs (hunk 2 above)
Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs:267]: @dannybtsai there are two possibilities for the 'tid' claim, therefore there are two constants.

**Review by brentschmaltz — APPROVED** [review]

File: test/Microsoft.IdentityModel.Validators.Tests/MicrosoftIdentityIssuerValidatorTest.cs (hunk 1 above)
Comment (brentschmaltz) [reply test/Microsoft.IdentityModel.Validators.Tests/MicrosoftIdentityIssuerValidatorTest.cs:325]: @jennyf19 using SecurityTokenDescriptor does not imply a JWE. Only if the EncryptingCredentials are set will a JWE be created.
You can use this property on SecurityTokenDescriptor to add additional payload claims: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/669515a4ab1e3201768e0d74cd5097db0ba4152b/src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs#L88

Comment (brentschmaltz) [reply test/Microsoft.IdentityModel.Validators.Tests/MicrosoftIdentityIssuerValidatorTest.cs:325]: Add this property to add additional header claims: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/669515a4ab1e3201768e0d74cd5097db0ba4152b/src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs#L97

**Review by GeoK — APPROVED** [review]

**Pull request merged**